.circleci/config.yml

```diff
@@ -269,31 +229,66 @@ jobs:
               -destination 'platform=iOS Simulator,name=iPhone 11' \
               -scheme ApplePlatformsIntegrationMobileTests
           working_directory: test/ApplePlatformsIntegrationTestApp
+
+  build-apple-runtime:
+    <<: *apple_defaults
+    steps:
+      - checkout
+      - restore_cache:
+          key: v3-repo-{{ .Environment.CIRCLE_PR_NUMBER }}
+      - run:
+          name: Set up workspace
+          command: mkdir -p /tmp/hermes/output
+      - run:
+          name: Install dependencies
+          command: |
+            brew install cmake
+            sudo gem install cocoapods
+      - run:
+          name: Build the iOS frameworks
+          command: ./utils/build-ios-framework.sh
+      - run:
+          name: Build the Mac frameworks
+          command: ./utils/build-mac-framework.sh
+      - run:
+          name: Build the CLI for Mac
+          command: cmake --build ./build_macosx --target github-cli-release
```

Comment:
This replaces entire `macos` job: https://github.com/facebook/hermes/pull/393/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L198

Reply:
In a future PR, I'd like to split this into `build-cli-tools.sh` for the CLI itself and just use plain Bash to build and package the CLI into `tgz` in a `package_apple_platforms` job. 

Unfortunately, this will take a bit more time. It is also not necessary to be implemented in this PR. Leaving it for the future.